spec/index.adoc

```diff
@@ -1471,15 +1473,15 @@ For each canister, the IC keeps track of “certified data”, a canister-define
 +
 The canister can update the certified data with this call. The passed data must be no larger than 32 bytes. This can be used any number of times.
 
-When executing a query method via a query call (i.e. in non-replicated state), the canister can fetch a certificate that authenticates to third parties the value last set via `ic0.certified_data_set`.
+When executing a query or composite query method via a query call (i.e. in non-replicated state), the canister can fetch a certificate that authenticates to third parties the value last set via `ic0.certified_data_set`.
```

Comment:
For composite queries, this is only currently supported in the root canister of the query call graph.

Reply:
> currently

Is it going to change soon? It's not difficult to model, but it would clutter the pseudocode below.

Reply:
I guess that is implicit in “via a query call”, but could be made clearer indeed.

Reply:
The pseudocode below gives a certificate to every call of a (composite) query method, not just the one called by the user.

Reply:
Getting certified variables work in the non-root canister isn't that easy, as some of the code has to be restructured. Some voices think that it's a simple rewrite, others think it might be more tricky.

Probably we need to try it and see how much work it is before we can give a timeline.

I'm fine with keeping the pseudocode simple, for now.

Reply:
+1 to making the ceritificate available everywhere in the call graph and keeping the spec simple

Reply:
Given that we want to release composite queries soon and the certificate is not available everywhere in the call graph, I've made this commit (https://github.com/dfinity/interface-spec/pull/144/commits/f1ec4cd267c9a6cb103598f73bc5b9d832c54cde) to specify the current behavior. Could you please double check? @ulan